File: 프로그래머스/lv1/이재표/둘만의 암호.java
import java.util.*;
class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        char[]alp=new char[26];
        for(int i=0;i<26;i++){
            alp[i]=((char)('a'+i));
        }
        for(int i=0;i<skip.length();i++){
            alp[skip.charAt(i)-'a']='-';
        }
        StringBuilder sb=new StringBuilder();
        for(int i=0;i<s.length();i++){
            int origin=s.charAt(i)-'a';
            int cnt=index;
            int idx=1;
            int changeIdx = 0;
            while(cnt>0){
                changeIdx=(origin+idx)%26;
                if(alp[changeIdx]!='-'){
                    idx++;
                    cnt--;
                    continue;
                }
                idx++;
            }
            sb.append(alp[changeIdx]);
        }
        return sb.toString();
    }
}

File: 프로그래머스/lv1/김원래/둘만의 암호.cpp
#include <string>
#include <vector>

using namespace std;

char _skip[26];

string solution(string s, string skip, int index) {
    string answer = "";
    
    for(int cnt = 0 ; cnt < skip.size() ; ++cnt)
    {
        int idx = skip[cnt] % 'a';
        _skip[idx] = 1;
    }
    

    for(int cnt = 0 ; cnt < s.size() ; ++cnt)
    {
        int checkIdx = s[cnt] % 'a';
        
        for(int _index = 1 ; _index <= index ; ++_index)
        {
            checkIdx = (checkIdx + 1) % 26;
            while(_skip[checkIdx] == 1) checkIdx = (checkIdx + 1) % 26;
        }
        
        answer.push_back(checkIdx + 'a');
    }
    
    return answer;
}

/*


넘어가는 것을 생각해서 모듈러 연산 + 남은걸 더하는 방식으로 간다?
배열로 만들어서 skip을 표시한다.


*/